StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/CustomSheet/CustomSheetLayoutView.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  CustomSheetLayoutView.swift
+//  StripeFinancialConnections
+//
+//  Created by Krisjanis Gaidis on 6/5/23.
+//
+
+import Foundation
+@_spi(STP) import StripeUICore
+import UIKit
+
+/// Reusable view that separates panes into two parts:
+/// 1. A scroll view for content
+/// 2. A footer that is "locked" and does not get affected by scroll view
+///
+/// Purposefully NOT a `UIView` subclass because it should only be used via
+/// `addToView` helper function.
+final class CustomSheetLayoutView {
+
+    private weak var scrollViewContentView: UIView?
+    private let paneLayoutView: UIView
+    let scrollView: UIScrollView
+
+    init(contentView: UIView, footerView: UIView?) {
+        self.scrollViewContentView = contentView
+
+        let scrollView = ResizableScrollView()
+        self.scrollView = scrollView
+        scrollView.addAndPinSubview(contentView)
+
+        let verticalStackView = HitTestStackView(
+            arrangedSubviews: [
+                CreateHandleView(),
+                scrollView,
+            ]
+        )
+        verticalStackView.spacing = 0
+        verticalStackView.axis = .vertical
+
+        if let footerView = footerView {
+            verticalStackView.addArrangedSubview(footerView)
+        }
+
+        self.paneLayoutView = verticalStackView
+    }
+
+    func addTo(view: UIView) {
+        // this function encapsulates an error-prone sequence where we
+        // must add `paneLayoutView` (and all it's subviews) to the `view`
+        // BEFORE we can add a constraint for `UIScrollView` content
+        view.addAndPinSubviewToSafeArea(paneLayoutView)
+        scrollViewContentView?.widthAnchor.constraint(equalTo: view.safeAreaLayoutGuide.widthAnchor).isActive = true
+    }
+}
+
+private final class ResizableScrollView: UIScrollView {
```

Comment:
This helps to creates a `UIScrollView` that will only start scrolling if the content height is greater than available parent height

The implementation is surprisingly simple. Arrived at it from iteration. `UIStackView` auto-magically handles this across multiple iOS versions.